ingest/Snakefile

```diff
@@ -0,0 +1,389 @@
+                --date-fields {params.date_fields} \
+                --expected-date-formats {params.expected_date_formats} \
+                --failure-reporting silent \
+            | augur curate titlecase \
+                --titlecase-fields {params.titlecase_fields} \
+                --articles {params.articles} \
+                --abbreviations {params.abbreviations} \
+            | augur curate apply-record-annotations \
+                --annotations {input.annotations} \
+                --id-field {params.annotations_id} \
+                --output-metadata {output.metadata_curate} 2>> {log}
+        """
+
+checkpoint filter_subsample:
+    input:
+        metadata_curate = "data/metadata_curate.tsv"
+    output:
+        metadata_subsample = "data/metadata_subsample.tsv"
+    log:
+        "logs/filter_subsample.txt",
+    benchmark:
+        "benchmarks/filter_subsample.txt"
+    conda:
+        "envs/nextstrain.yaml"
+    shell:
+        """
+        augur filter \
+            --metadata {input.metadata_curate} \
+            --metadata-id-columns accession \
+            --query "(mbases > 180 & (country != 'Uncalculated'))" \
+	        --group-by country year \
+	        --sequences-per-group 1 \
+	        --min-date 2024 \
+	        --output-metadata {output.metadata_subsample}
+        """
+
+rule run_tbprofiler:
+    output:
+        "data/tbprofiler/results/{sample}.results.json",
+        touch("data/tbprofiler/flags/{sample}_flag.txt")
+    params:
+        outdir="data/fastq",
+        tb_outdir="data/tbprofiler",
+        threads=4,
+        s3_bucket=config["s3_bucket"],
+        output_path="data/tbprofiler/results/{sample}.results.json"
+    conda:
+        "envs/tb-profiler.yaml"
+    shell:
+        r'''
+        set -x  # <--- DEBUGGING
+        
+        s3_path="files/workflows/tb/{params.output_path}"
+
+        if ! command -v aws &> /dev/null; then
+            echo "Error: AWS CLI is not available." >&2
+            exit 1
+        fi
+
+        if ! aws s3 ls "s3://{params.s3_bucket}" > /dev/null 2>&1; then
+            echo "Error: Unable to access s3://{params.s3_bucket}." >&2
+            exit 1
+        fi
+
+        if aws s3api head-object --bucket {params.s3_bucket} --key "$s3_path" > /dev/null 2>&1; then
+            echo "Found tb-profiler results on S3. Downloading to {params.output_path} …" >&2
+            mkdir -p $(dirname {params.output_path})
+            aws s3 cp "s3://{params.s3_bucket}/$s3_path" {params.output_path}
+        else
+            fastq1={params.outdir}/{wildcards.sample}_1.fastq.gz
+            fastq2={params.outdir}/{wildcards.sample}_2.fastq.gz
+
+            if [[ ! -f "$fastq1" ]]; then
+                echo "Downloading fastq files…" >&2
+                parallel-fastq-dump --split-files --gzip \
+                    --sra-id {wildcards.sample} --threads {params.threads} \
+                    --outdir {params.outdir}
+                exit 1
```

Comment:
This `exit 1` is the error behind the `20609186.log` (from slack) - I think this is unintentional (as in, we don't want to exit we want to continue with the rest of the bash commands). This part is repeated in `run_snippy` bash code as well.

P.S. the line in your log file
```
+ exit 1
```

Is bash's `set -x` printing out each line to STDERR before it runs, i.e. we're about to run the command `exit 1` which will cause the snakemake rule to fail.